languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/GermanSpellerRule.java

```diff
@@ -2213,20 +2213,24 @@ protected boolean ignorePotentiallyMisspelledWord(String word) throws IOExceptio
       if ((part1.matches(".*(heit|keit|ion|ität|schaft|ung|tät)s") || wordsNeedingInfixS.contains(part1noInfix)) &&
           isNoun(part2uc)) {
         if (part1noInfix.matches("Action|Session|Champion|Jung|Wahrung") ||
-            part2uc.matches("First|Frist|Firsten|Fristen") ||  // too easy to mix up
+            part2uc.matches("First|Firsten") ||  // too easy to mix up 
             part1.endsWith("schwungs") || part1.endsWith("sprungs") || isMisspelled(part1noInfix) || isMisspelled(part2uc)) {
           return false;
         }
         return true;
       }
     }
+    String part2uc = uppercaseFirstChar(part2);
     if (!hasInfixS && part1.length() >= 3 && part2.length() >= 4 && !part2.contains("-") && startsWithLowercase(part2) &&
-        wordsWithoutInfixS.contains(part1) && !isMisspelled(part1) &&
+        (wordsWithoutInfixS.contains(part1) || 
+        (part1.matches(".*(mus|ss|z)") && isNoun(part2uc)) || isPluralNoun(part1) && part1.endsWith("en") && 
```

Comment:
The logic conditions are difficult to read now, can we move parts of them to their own method or find a different solution to make them easier to read?